README.md

```diff
@@ -8,21 +8,193 @@
+
+      // Send the token request
+      json|error response = hubendpoint->post("/oauth/v1/token", payload, {
+         "Content-Type": "application/x-www-form-urlencoded"
+      });
+
+      return response;
+
+   };
+
+   service / on new http:Listener(9090) {
+    resource function get .(http:Caller caller, http:Request req) returns error? {
+        // Extract the "code" query parameter from the URL
+        string? code = req.getQueryParamValue("code");
+
+        if code is string {
+            // Log the received code
+            io:println("Authorization code received: " + code);
+            // Respond to the client with the received code
+            check caller->respond("Received code: " + code);
+            json|error res = getRefresh(code);
+            io:println(res);
+        } else {
+            // Respond with an error message if no code is found
+            check caller->respond("Authorization code not found.");
+        }
+    }
+   };
+   ```
+
+3. Paste it in the browser and select your developer test account to intall the app when prompted.
+    
+   <img src="docs/resources/install_app.png" alt="Hubspot app creation 1 testacc3" style="width: 70%;">
+
+4. After the installation, you will be redirected to the localhost server you started in the previous step. The server will print the authorization code in the console.
+
+   ```json
+   {
+     "token_type": "bearer",
+     "refresh_token": "<Refresh Token>",
+     "access_token": "<Access Token>",
+     "expires_in": 1800
+   }
+   ```
+
+5. Store the `refresh_token` securely for use in your application
 
 ## Quickstart
 
-[//]: # (TODO: Add a quickstart guide to demonstrate a basic functionality of the module, including sample code snippets.)
+To use the `HubSpot Deals` connector in your Ballerina application, update the `.bal` file as follows:
+
+### Step 1: Import the module
+
+Import the `hubspot.crm.obj.deals` module.
+
+```
+import ballerinax/hubspot.crm.obj.deals;
+```
+
+### Step 2: Instantiate a new connector
+
+1. Create a `config.toml` file and, configure the obtained credentials in the above steps as follows:
+   ```toml
+   clientId = "<Client ID>"
+   clientSecret = "<Client Secret>"
+   refreshToken = "<Access Token>"
+   ```
+
+2. Create a `deals:ConnectionConfig` with the obtained access token and initialize the connector with it.
+
+   ```ballerina
+   //auth confguration for hubspot
+   deals:OAuth2RefreshTokenGrantConfig auth = {
+      clientId: clientId,
+      clientSecret: clientSecret,
+      refreshToken: refreshToken,
+      credentialBearer: oauth2:POST_BODY_BEARER
+      };
+
+   deals:ConnectionConfig config = {auth: auth};
+   //authorized http client to access hubspot
+   final deals:Client hubspot = check new deals:Client(config);
+   ```
+
+### Step 3: Use Connector Operations
```

Comment:
Let's update this section as well. You can check the Ballerina zendesk connector as a reference.

https://github.com/ballerina-platform/module-ballerinax-zendesk/blob/main/ballerina/Package.md#step-3-invoke-the-connector-operation